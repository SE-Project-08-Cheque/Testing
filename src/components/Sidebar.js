import React from 'react';
import {
  Flex,
  Box,
  Heading,
  Button,
  useColorMode,
  IconButton,
  useDisclosure,
  ReactRouterLink,
  Center,
  Text,
  Square,
  VStack,
  StackDivider,
  Table,
  Td,
  Tr,
  Thead,
  Th,
  Tbody,
  TableCaption,
  CircularProgress,
  Badge,
  Progress,
  Divider,
} from '@chakra-ui/react';
import { Avatar, AvatarBadge } from '@chakra-ui/react';
import { Link } from 'react-router-dom';
import SessionService from '../Services/SessionService';

// sidebar nav config

function Sidebar() {


  return (
    <>
      <Flex
        paddingTop="10px"
        h="800px"
        left="0px"
        top="0px"
        width={{ base: '100%', md: '300px' }}
        height={{ base: '520px', md: '100%' }}
        borderTop="1px"
        borderTopColor="gray.600"
        borderTopWidth="90px"
        position="fixed"
        rounded="md"
        backgroundColor="gray.700"
        zIndex="30"
      >
        <Box
          width="100%"
          borderStyle="solid"
          borderColor="gray.600"
          borderWidth="0px"
        >

          <Box h="50px" mb="70px">
            <Center>
              <Text color="white">{"" + ""}</Text>
              {/* {JSON.parse(SessionService.getdata()).full_name} */}
            </Center>
            <Center mb="10px">
              <Badge variant="solid" colorScheme="green">
                Online
              </Badge>
            </Center>
            <Progress
              ml="50px"
              mr="50px"
              colorScheme="green"
              size="xs"
              isIndeterminate
            />
          </Box>

          <Box width="100%">
            <VStack
              divider={<StackDivider borderColor="gray.500" />}
              spacing={5}
              align="stretch"
            >



              <a href="/home">
                <Button
                  h="50px"
                  w="100%"
                  data-testid = 'overview'
                  _hover={{
                    background: 'gray.800',
                    color: 'white',
                    transitionDuration: '0.6s',
                  }}
                  rounded="box"
                  boxShadow="dark-lg"
                  colorScheme="gray.600"
                >
                  Overview
                </Button>
              </a>
              <a href="/Account_Details">
                <Button
                  h="50px"
                  w="100%"
                  data-testid = 'personal_details'
                  _hover={{
                    background: 'gray.800',
                    color: 'white',
                    transitionDuration: '0.6s',
                  }}
                  rounded="box"
                  boxShadow="dark-lg"
                  colorScheme="gray.600"
                >
                  <Center>Personal Details</Center>
                </Button>
              </a>

              <a href="/Contact_Details">
                <Button
                  h="50px"
                  w="100%"
                  data-testid = 'contact_details'
                  _hover={{
                    background: 'gray.800',
                    color: 'white',
                    transitionDuration: '0.6s',
                  }}
                  rounded="box"
                  boxShadow="dark-lg"
                  colorScheme="gray.600"
                >
                  Contact Details
                </Button>
              </a>

              <a href="/AllNotifications">
                <Button
                  h="50px"
                  w="100%"
                  data-testid = 'notification'
                  _hover={{
                    background: 'gray.800',
                    color: 'white',
                    transitionDuration: '0.6s',
                  }}
                  rounded="box"
                  boxShadow="dark-lg"
                  colorScheme="gray.600"
                >
                  <Center>Notification</Center>
                </Button>
              </a>

              <a href="/ForgotPin">
                <Button
                  h="50px"
                  w="100%"
                  data-testid = 'forgot_pin'
                  _hover={{
                    background: 'gray.800',
                    color: 'white',
                    transitionDuration: '0.6s',
                  }}
                  rounded="box"
                  boxShadow="dark-lg"
                  colorScheme="gray.600"
                >
                  <Center>Fogot Pin</Center>
                </Button>
              </a>


              <Divider orientation="horizontal" />
            </VStack>
          </Box>
        </Box>
      </Flex>
    </>
  );
}

export default Sidebar;
