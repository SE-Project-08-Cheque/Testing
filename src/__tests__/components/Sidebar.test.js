import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from '../../components/Sidebar';
import {render, cleanup, fireEvent} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Sidebar />, div)
})

it("renders button correctly", () => {
    const {getByTestId} = render(<Sidebar />)
    expect(getByTestId('overview')).toBeTruthy();
})

it("matches snapshot", () => {
    const tree = renderer.create(<Sidebar />).toJSON();
    expect(tree).toMatchSnapshot();
})

// it("checkButtonRender", () => {
//     const {QueryByTitle} = render(<Sidebar />)
//     expect(queryByTitle("asd")).toBeTruthy(); 
// })

// describe("ClickButton", () => {
//     it("onClick", () => {
//         const {QueryByTitle} = render(<Sidebar />)
//         expect(queryByTitle("asd").innerHTML).toHaveTextContent("PLACE HERE"); 
//         const btn = queryByTitle("asd");
// fireEvent.click(btn)
//         expect(queryByTitle("asd").innerHTML).toHaveTextContent("yOU Clicked"); 
//     })
// })

// describe("changeInInput", () => {
//     it("onChange", () => {
//         const {QueryByTitle} = render(<Sidebar />)
//         const input = queryByTitle("asd");
//         fireEvent.change(input, {target: {value: "testValue"}}); //change input to testValue. Like we gave an input
//         expect(input.value).toBe("testValue");
//     })
// })