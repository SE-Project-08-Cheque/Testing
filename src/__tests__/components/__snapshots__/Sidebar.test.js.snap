// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-ure0nq"
>
  <div
    className="css-dbpzvv"
  >
    <div
      className="css-1rul54"
    >
      <div
        className="css-gmuwbf"
      >
        <p
          className="chakra-text css-kbn7if"
        >
          
        </p>
      </div>
      <div
        className="css-sc5c3h"
      >
        <span
          className="chakra-badge css-ifsvay"
        >
          Online
        </span>
      </div>
      <div
        className="css-16n1mrt"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          className="css-12om7f"
          data-indeterminate=""
          role="progressbar"
          style={
            Object {
              "width": "0%",
            }
          }
        />
      </div>
    </div>
    <div
      className="css-8atqhb"
    >
      <div
        className="chakra-stack css-zsoya5"
      >
        <a
          href="/home"
        >
          <button
            className="chakra-button css-35wvkn"
            data-testid="overview"
            type="button"
          >
            Overview
          </button>
        </a>
        <div
          className="chakra-stack__divider css-1kh2tyg"
        />
        <a
          href="/Account_Details"
        >
          <button
            className="chakra-button css-35wvkn"
            type="button"
          >
            <div
              className="css-gmuwbf"
            >
              Personal Details
            </div>
          </button>
        </a>
        <div
          className="chakra-stack__divider css-1kh2tyg"
        />
        <a
          href="/Contact_Details"
        >
          <button
            className="chakra-button css-35wvkn"
            type="button"
          >
            Contact Details
          </button>
        </a>
        <div
          className="chakra-stack__divider css-1kh2tyg"
        />
        <a
          href="/AllNotifications"
        >
          <button
            className="chakra-button css-35wvkn"
            type="button"
          >
            <div
              className="css-gmuwbf"
            >
              Notification
            </div>
          </button>
        </a>
        <div
          className="chakra-stack__divider css-1kh2tyg"
        />
        <a
          href="/ForgotPin"
        >
          <button
            className="chakra-button css-35wvkn"
            type="button"
          >
            <div
              className="css-gmuwbf"
            >
              Fogot Pin
            </div>
          </button>
        </a>
        <div
          className="chakra-stack__divider css-1kh2tyg"
        />
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-1upb9tn"
          role="separator"
        />
      </div>
    </div>
  </div>
</div>
`;
